pkg/status/status.go

```diff
@@ -200,3 +227,71 @@ func (s *status) updateStatus() {
 		log.Fatalf("Error %v updating ClusterOperator", err)
 	}
 }
+
+// syncChannelConsumer will listen on the sync channel and update the status'
+// syncs and successfulSyncs fields until the stopCh is closed.
+func syncChannelConsumer(instance *status, stopCh <-chan struct{}) {
+	log.Info("Starting sync consumer")
+	go wait.Until(func() {
+		// Listen on SyncCh and update instance accordingly
+		err := <-syncCh
+		instance.syncs++
+		if err == nil {
+			instance.successfulSyncs++
+		}
+		log.Infof("Successful Syncs / Total Syncs: %d/%d", instance.successfulSyncs, instance.syncs)
+
+	}, 0*time.Second, stopCh)
+}
+
+// monitorClusterStatus updates the clusteroperator's status based on
+// the number of successful syncs / total syncs
+func (s *status) monitorClusterStatus(stopCh <-chan struct{}) {
+	// exit if cluster operator api is not present
+	if s.coAPINotPresent {
+		return
+	}
+
+	go syncChannelConsumer(instance, stopCh)
+	go wait.Until(func() {
+		// create the cluster operator in an initial state if it does not exist
+		if s.clusterOperator == nil {
+			s.setProgressing(fmt.Sprintf("Progressing towards %s", s.version))
+		}
+
+		// wait until the operator has synced enough objects
+		if s.syncs < minSyncsBeforeReporting {
+			log.Printf("Waiting to observe more syncs")
+			return
+		}
+
+		// update the status with the appropriate state
+		if s.succeeding() {
+			s.setAvailable(fmt.Sprintf("%s is available", s.version))
+			// if we've reported success, we can sleep longer, otherwise we want to keep watching for
+			// successful
+			time.Sleep(5 * time.Minute)
+		} else {
+			s.setFailing(fmt.Sprintf("Waiting for updates to take effect"))
```

Comment:
Are we waiting for an update or for the sync ratio to be achieved?

Reply:
This is a good catch. At this point all sync reports should be successful unless Marketplace encounters a critical error. If we set this to failing, we should report that marketplace has encountered a critical failure.